pyccel/parser/semantic.py

```diff
@@ -1426,7 +1429,7 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
                 # ...
                 # Add memory deallocation for array variables
                 # check if the variable is an ndarray and not astacked variable and that is a view
-                if lhs.is_ndarray and not lhs.is_stack_array and not (lhs.is_pointer and lhs.cls_base):
+                if lhs.is_ndarray and not lhs.is_stack_array and not (lhs.is_pointer and not lhs.is_view):
```

Comment:
The expression `not (lhs.is_pointer and not lhs.is_view)` might be clearer as `(lhs.is_view or not lhs.is_pointer)`